.github/workflows/coverity.yml

```diff
@@ -35,6 +35,7 @@ jobs:
       run: cov-analysis-linux64/bin/cov-build --dir cov-int make
       env:
         PYTHON_USER_SPACE: 1
+        PYTHON_EDITABLE_BUILD: 1
```

Comment:
I think the job that actually require this for sure is CodeCoverage. Or at least, it was introduced in the codecoverage job that is part of Nightly.yml. But then unsure if it's worth also elsewhere.

Reply:
I could find no comments explaining why we need an editable build, I saw this yml file was changed so I guessed this was the relevant job.

I'm fine to add it where needed, let me know

Reply:
I don't properly know either, I know that before that PR CodeCoverage was not building and after it was, but not sure if the --editable part had anything to do with it.

Reply:
Editable builds are only useful for local development - rather than copying the package files into python install, it uses a symlink, so CI shouldn't need them at all

Reply:
Should we then keep the editable builds around at all? Or are they useful and is the current change to make them opt-in sufficient?

Reply:
@Mause are the editable builds useful?
I am just trying to preserve the behavior you introduced

Reply:
I have found one case where it was needed, I was building with python3.7.12 and I needed `--editable` otherwise the `package_build.py` module could not be found:
```py
    sys.path.append(os.path.join(script_path, '..', '..', 'scripts'))
    import package_build
```

this expects the file to be in the same location, but without --editable the setup.py and the rest of the project can get copied to another location.

Reply:
Let's keep it in then